engine/src/scripts/environment/agent/Agent.cs

```diff
@@ -61,7 +61,9 @@ public int Id
     
     protected void UpdateEnergy(double delta)
     {
-        this.energy -= Config.Instance.Environment.EnergyLossPerSecond * (float)delta;
+        this.energy -= Config.Instance.Environment.EnergyBaseLossPerSecond * (float)delta 
```

Comment:
The current implementation does not properly normalize energy loss due to movement, resulting in an incorrect config value. For instance an agent with a maximum speed of 400.0 should lose full 0.2 energy per second according to the config value. However it will lose: ``0.2 (from config) * 400 (agent speed) * 0.016 (delta) * 60 (frames per second) = 0.2 * 400 = 80``. To fix this I would normalize the agent's speed to 100 units and then multiply the result by the energy loss from config. This way, the config value will correspond to the energy loss for an agent moving at a speed of 100 units per second. And using previous values it should lose 0.8 energy per second, because it loses 0.2 energy per second per 100 units of speed.
```suggestion
        float energyLoss = Config.Instance.Environment.EnergyBaseLossPerSecond
                           + Config.Instance.Environment.EnergyLossPerSecondPer100UnitsOfMovement * (this.Velocity.Length()/100.0f)
                           + Config.Instance.Environment.EnergyLossPerSecondTurn * Mathf.Abs(this.currentRotation);
        this.energy -= energyLoss * (float)delta;
```